### ISSUE TITLE
Problem with large repository

### ISSUE THREAD

**Issue by noornee** [hacktivist123/goignore#27]
We have a large repository; about (~ 11MB). 

# Screenshot
![goignore_large_repo](https://github.com/hacktivist123/goignore/assets/71889751/f9739b00-7a72-40d6-b538-818a229a9681)

This is because the binary file for goignore was accidentally? committed and even though its been deleted already, it's still in the github history.

I tried looking for way to remove a large file from a github history and most of what i found were kind of tricky for me to wrap my head around until i came accross this [article](https://www.phase2technology.com/blog/removing-large-files-git-bfg)

what do you think?  @hacktivist123 @quamejnr @chuksgpfr @AdeboyeDN 
is there a better way to do this? 

p.s. I created a random private repo to test this and it worked. I cant tell if its not going to cause some issues for a repo like this with forks..

**Comment by chuksgpfr** [comment]
There's a way to delete it from the ~HEAD. I'll do it when I'm on my laptop.

**Comment by noornee** [comment]
> There's a way to delete it from the ~HEAD. I'll do it when I'm on my laptop.

alright, great!
please when you do, i would love to see the git command you used. Tyyy ^^

**Comment by chuksgpfr** [comment]
First i need to find the commit

**Comment by chuksgpfr** [comment]
This [commit](https://github.com/hacktivist123/goignore/commit/a3b43fb12272f063681953567898342523c51862) committed a binary file

**Comment by hacktivist123** [comment]
Damn, that was me, I thought I deleted the binary. Please let me know how you deleted it if you do

**Comment by noornee** [comment]
> This [commit](https://github.com/hacktivist123/goignore/commit/a3b43fb12272f063681953567898342523c51862) committed a binary file

@chuksgpfr  The [first commit](https://github.com/hacktivist123/goignore/commit/c0728f279570bc38d29f7ead84855b3851d86565#diff-28c68788b1b3fc84178345c3a8030e1eff030ea4caeed8c38af9855daab9338a) also has a binary file committed to it

**Comment by hacktivist123** [comment]
still doing this @chuksgpfr ?

**Comment by chuksgpfr** [comment]
Yeah. I'll get it done today.

**Comment by hacktivist123** [comment]
Thanks my boss

**Comment by chuksgpfr** [comment]
I am back here. @noornee how did you see the size of the repo? I got 245KB

**Comment by noornee** [comment]
> I am back here. @noornee how did you see the size of the repo?

Welcome!
I cloned and checked the size of the goignore directory with [du](https://www.geeksforgeeks.org/du-command-linux-examples)


# Screenshot
![goignore_large_repo](https://github.com/hacktivist123/goignore/assets/71889751/f9739b00-7a72-40d6-b538-818a229a9681)

 >... I got 245KB

strange. how did you check the size? 
or have you successfully run the git commands to purge the binary files from goignore's history ?

**Comment by chuksgpfr** [comment]
Different sizes. The size that was downloaded from GitHub using this command `curl https://github.com/hacktivist123/goignore > /dev/null | grep size | tr -dc '[:digit:]'` is different.

**Comment by hacktivist123** [comment]
What are the sizes?

**Comment by noornee** [comment]
> Different sizes. The size that was downloaded from GitHub using this command `curl https://github.com/hacktivist123/goignore > /dev/null | grep size | tr -dc '[:digit:]'` is different.

i dont think this command would give you the size of the github repo. it just downloads the webpage and prints the lines that matchs the `size` pattern and essentially outputs the digits.

## ss
![invalid](https://github.com/hacktivist123/goignore/assets/71889751/c7aaa2af-a78e-40b2-8234-c73c3bcaf1d6)


you could access the size of the repo using the github api.

i.e.
```bash
curl --silent https://api.github.com/repos/hacktivist123/goignore | grep size
```
p.s. the size it would give you would be in kilobytes. but its approximately 11MB

**Comment by noornee** [comment]
> What are the sizes?

you could check the size of the repo with a GUI filemanager. the size is approx 11MB.

## SCREENSHOT

![thunar_goignore_size](https://github.com/hacktivist123/goignore/assets/71889751/665f94d3-e84b-4908-b877-b3bc00285001)
